brpc-java-core/src/main/java/com/baidu/brpc/client/RpcClient.java

```diff
@@ -383,20 +380,29 @@ public void removeChannel(Channel channel) {
     }
 
     public <T> AsyncAwareFuture<T> sendRequest(Request request) {
-        Channel channel = request.getChannel();
+        RpcFuture rpcFuture = new RpcFuture();
+        // generate correlationId
+        long correlationId = FastFutureStore.getInstance(0).put(rpcFuture);
+        request.setCorrelationId(correlationId);
+        request.retain();
```

Comment:
request.retain() should be called before writeAndFlush(), after encodeRequest()

Reply:
fixed